rasterio/_warp.pyx

```diff
@@ -652,56 +659,76 @@ def _calculate_default_transform(src_crs, dst_crs, width, height,
+    # We use an in-memory VRT dataset.
+    vrt_doc = _suggested_proxy_vrt_doc(
+        width,
+        height,
+        transform=transform,
+        crs=src_crs,
+        gcps=gcps
+    ).decode('ascii')
+    hds = open_dataset(vrt_doc, 0x00 | 0x02 | 0x04, ['VRT'], {}, None)
 
     try:
         imgProjOptions = CSLSetNameValue(imgProjOptions, "GCPS_OK", "TRUE")
         imgProjOptions = CSLSetNameValue(imgProjOptions, "MAX_GCP_ORDER", "0")
         imgProjOptions = CSLSetNameValue(imgProjOptions, "DST_SRS", wkt)
+
         for key, val in kwargs.items():
             key = key.upper().encode('utf-8')
+
             if key == b"RPC_DEM":
-                # don't .upper() since might be a path
+                # don't .upper() since might be a path.
                 val = str(val).encode('utf-8')
             else:
                 val = str(val).upper().encode('utf-8')
-            imgProjOptions = CSLSetNameValue(
-                imgProjOptions, <const char *>key, <const char *>val)
-            log.debug("Set _calculate_default_transform Transformer option {0!r}={1!r}".format(key, val))
-        hds = open_dataset(vrt_doc, 0x00 | 0x02 | 0x04, ['VRT'], {}, None)
+
+            imgProjOptions = CSLSetNameValue(imgProjOptions, <const char *>key, <const char *>val)
+            log.debug("Set image projection option {0!r}={1!r}".format(key, val))
+
         if rpcs:
             if hasattr(rpcs, 'to_gdal'):
                 rpcs = rpcs.to_gdal()
+
             for key, val in rpcs.items():
                 key = key.upper().encode('utf-8')
                 val = str(val).encode('utf-8')
-                papszMD = CSLSetNameValue(
-                    papszMD, <const char *>key, <const char *>val)
+                papszMD = CSLSetNameValue(papszMD, <const char *>key, <const char *>val)
+
             exc_wrap_int(GDALSetMetadata(hds, papszMD, "RPC"))
             imgProjOptions = CSLSetNameValue(imgProjOptions, "SRC_METHOD", "RPC")
 
         hTransformArg = exc_wrap_pointer(
             GDALCreateGenImgProjTransformer2(hds, NULL, imgProjOptions)
         )
-        exc_wrap_int(
-            GDALSuggestedWarpOutput2(
-                hds, GDALGenImgProjTransform, hTransformArg,
-                geotransform, &npixels, &nlines, extent, 0))
 
-        log.debug("Created transformer and warp output.")
+        try:
+            # This function may put errors on GDAL's error stack while
+            # still returning 0 (no error). Thus we always check and
+            # clear the stack and treat errors as warnings.
+            retval = GDALSuggestedWarpOutput2(
+                hds,
+                GDALGenImgProjTransform,
+                hTransformArg,
+                geotransform,
+                &npixels,
+                &nlines,
+                extent,
+                0
+            )
+
+            _ = exc_wrap_int(retval >= 0)
+
+        except CPLE_AppDefinedError as err:
+            if retval == 0:
```

Comment:
I put this call and return value handler/checker in its own try/except block. Inside of checking exception message contents, we discard exceptions if GDALSuggestedWarpOutput2 succeeded, i.e., returned `0`. @rouault is there any downside to doing this?

Reply:
> Instead of checking exception message contents we discard all exceptions if GDALSuggestedWarpOutput2 succeeded, i.e., returned 0. @rouault is there any downside to doing this?

none that I can't think of